### PR TITLE
build: Centralize rust-vmm crates to workspace

### PR DESCRIPTION
**Pull request by TimePrinciple**
### Summary of the PR

- build: Centralize rust-vmm crates to workspace

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR have Signed-Off-By trailers (with
  `git commit -s`), and the commit message has max 60 characters for the
  summary and max 75 characters for each description line.
- [ ] All added/changed functionality has a corresponding unit/integration
  test.
- [ ] All added/changed public-facing functionality has entries in the "Upcoming 
  Release" section of CHANGELOG.md (if no such section exists, please create one).
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Comment by TimePrinciple** [comment]
Need rust-vmm/vm-virtio#312 to be merged and release :)

**Comment by TimePrinciple** [comment]
I think we have to re-trigger our CI :)

**Comment by stefano-garzarella** [comment]
> I think we have to re-trigger our CI :)

I can re-trigger it, but did we make a new release of crates in vm-virtio?

**Comment by stefano-garzarella** [comment]
> > > I think we have to re-trigger our CI :)
> > 
> > 
> > I can re-trigger it, but did we make a new release of crates in vm-virtio?
> 
> Oh yes, I forgot it, should we release now 🤔

Yep, but I'm busy this week. If you have time, feel free to start opening an issue there, sending PRs to update changelog and Cargo.toml file with new version. When the PR is merged, I can do the tag and release it.

**Comment by TimePrinciple** [comment]
> > > > I think we have to re-trigger our CI :)
> > > 
> > > 
> > > I can re-trigger it, but did we make a new release of crates in vm-virtio?
> > 
> > 
> > Oh yes, I forgot it, should we release now 🤔
> 
> Yep, but I'm busy this week. If you have time, feel free to start opening an issue there, sending PRs to update changelog and Cargo.toml file with new version. When the PR is merged, I can do the tag and release it.

I'll handle this 😉

**Comment by TimePrinciple** [comment]
@stefano-garzarella @Ablu  PTAL when convenient :)

File: vhost-user-backend/CHANGELOG.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 ### Added
 
 ### Changed
+- [[#262]](https://github.com/rust-vmm/vhost/pull/262) Update virtio-queue
+dependency to 0.14.0 and vm-memory to 0.16.0; move rust-vmm crates to workspace.
```
Comment (Ablu) [inline vhost-user-backend/CHANGELOG.md:8]: entries below tended to ident the wrapped line to match the part after `- `.

**Review by Ablu — COMMENTED** [review]
change LGTM.

But I am a bit worried about dependabot... Does it support the workspace dependencies? https://github.com/dependabot/dependabot-core/issues/7896 reads like there are some problems?

**Comment by stefano-garzarella** [comment]
> change LGTM.
> 
> But I am a bit worried about dependabot... Does it support the workspace dependencies? [dependabot/dependabot-core#7896](https://github.com/dependabot/dependabot-core/issues/7896) reads like there are some problems?

Yeah, it was the same for me in vm-virtio: https://github.com/rust-vmm/vm-virtio/pull/312#issuecomment-2404260635
At the end we merged it, so maybe let's see how it goes with it before start to use that in other workspaces.

File: vhost-user-backend/CHANGELOG.md (hunk 1 above)
Comment (TimePrinciple) [reply vhost-user-backend/CHANGELOG.md:8]: Noted, and I suggest we use `mdformat` to take care of formatting :)

**Comment by stefano-garzarella** [comment]
> > > change LGTM.
> > > But I am a bit worried about dependabot... Does it support the workspace dependencies? [dependabot/dependabot-core#7896](https://github.com/dependabot/dependabot-core/issues/7896) reads like there are some problems?
> > 
> > 
> > Yeah, it was the same for me in vm-virtio: [rust-vmm/vm-virtio#312 (comment)](https://github.com/rust-vmm/vm-virtio/pull/312#issuecomment-2404260635) At the end we merged it, so maybe let's see how it goes with it before start to use that in other workspaces.
> 
> How about repurposing this PR to solely centralize crates, and trigger dependabot manually to see if it works :)

Good idea, IIUC I can do it via "Insights -> Dependency Graph -> Dependabot -> Cargo.toml -> Checks for updates", right?

**Comment by TimePrinciple** [comment]
> > > > change LGTM.
> > > > But I am a bit worried about dependabot... Does it support the workspace dependencies? [dependabot/dependabot-core#7896](https://github.com/dependabot/dependabot-core/issues/7896) reads like there are some problems?
> > > 
> > > 
> > > Yeah, it was the same for me in vm-virtio: [rust-vmm/vm-virtio#312 (comment)](https://github.com/rust-vmm/vm-virtio/pull/312#issuecomment-2404260635) At the end we merged it, so maybe let's see how it goes with it before start to use that in other workspaces.
> > 
> > 
> > How about repurposing this PR to solely centralize crates, and trigger dependabot manually to see if it works :)
> 
> Good idea, IIUC I can do it via "Insights -> Dependency Graph -> Dependabot -> Cargo.toml -> Checks for updates", right?

Good to go, let's have a try :)

**Review by stefano-garzarella — APPROVED** [review]

**Review by rbradford — APPROVED** [review]

**Comment by stefano-garzarella** [comment]
@Ablu @TimePrinciple it looks like dependabot is doing right with this PR merged. I triggered it manually and it created https://github.com/rust-vmm/vhost/pull/269

**Pull request merged**